### PR TITLE
refactor: clarify config

### LINKED ISSUE #28: Feature: clarify configurations for plugins
### Sources

Danbooru, Gelbooru, Konachan.com (NSFW), Konachan.net (SFW), Lolibooru, Lolicon, Pixiv, Safebooru, Yande.re

### Describe the problem related to the feature request

Configurations should be clarified and classified into several sections.

### Describe the solution you'd like

```text
全局设置
|-- label
|-- weight
搜索设置
|-- ...
```

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MaikoTan**
fix #28

### CODE REVIEW

File: packages/gelbooru/src/index.ts [hunk 1]
```diff
@@ -44,11 +44,15 @@ namespace GelbooruImageSource {
...
-    weight: Schema.number().min(1).default(1).description('图源权重。在多个符合标签的图源中，将按照各自的权重随机选择。'),
-    endpoint: Schema.string().description('gelbooru 的 URL。').default('https://gelbooru.com/index.php'),
-  })
+  export const Config: Schema<Config> = Schema.intersect([
+    Schema.object({
+      label: Schema.string().default('gelbooru').description('图源标签，可用于在指令中手动指定图源。'),
+      weight: Schema.number().min(1).default(1).description('图源权重。在多个符合标签的图源中，将按照各自的权重随机选择。'),
+    }).description('全局设置'),
+    Schema.object({
+      endpoint: Schema.string().description('gelbooru 的 URL。').default('https://gelbooru.com/index.php'),
```
Comment (MaikoTan) [inline packages/gelbooru/src/index.ts:53]:
```suggestion
      endpoint: Schema.string().description('Gelbooru 的 URL。').default('https://gelbooru.com/index.php'),
```

Use upper case if refer to the site itself

File: packages/konachan/src/index.ts [hunk 2]
```diff
@@ -39,14 +39,18 @@ namespace KonachanImageSource {
...
+  export const Config: Schema<Config> = Schema.intersect([
+    Schema.object({
+      label: Schema.string().default('konachan').description('图源标签，可用于在指令中手动指定图源。'),
+      weight: Schema.number().min(1).default(1).description('图源权重。在多个符合标签的图源中，将按照各自的权重随机选择。'),
+    }).description('全局设置'),
+    Schema.object({
+      endpoint: Schema.union([
+        Schema.const('https://konachan.com/').description('konachan.com (NSFW)'),
+        Schema.const('https://konachan.net/').description('konachan.net (SFW)')
+      ]).description('konachan 的 URL。').default('https://konachan.com/'),
```
Comment (MaikoTan) [inline packages/konachan/src/index.ts:51]:
```suggestion
      endpoint: Schema.union([
        Schema.const('https://konachan.com/').description('Konachan.com (NSFW)'),
        Schema.const('https://konachan.net/').description('Konachan.net (SFW)')
      ]).description('Konachan 的 URL。').default('https://konachan.com/'),
```

File: packages/lolibooru/src/index.ts [hunk 3]
```diff
@@ -41,11 +41,15 @@ namespace LolibooruImageSource {
...
-    weight: Schema.number().min(1).default(1).description('图源权重。在多个符合标签的图源中，将按照各自的权重随机选择。'),
-    endpoint: Schema.string().description('lolibooru 的 URL。').default('https://lolibooru.moe'),
-  })
+  export const Config: Schema<Config> = Schema.intersect([
+    Schema.object({
+      label: Schema.string().default('lolibooru').description('图源标签，可用于在指令中手动指定图源。'),
+      weight: Schema.number().min(1).default(1).description('图源权重。在多个符合标签的图源中，将按照各自的权重随机选择。'),
+    }).description('全局设置'),
+    Schema.object({
+      endpoint: Schema.string().description('lolibooru 的 URL。').default('https://lolibooru.moe'),
```
Comment (MaikoTan) [inline packages/lolibooru/src/index.ts:50]:
```suggestion
      endpoint: Schema.string().description('Lolibooru 的 URL。').default('https://lolibooru.moe'),
```

File: packages/safebooru/src/index.ts [hunk 4]
```diff
@@ -45,11 +45,15 @@ namespace SafebooruImageSource {
...
-    weight: Schema.number().min(1).default(1).description('图源权重。在多个符合标签的图源中，将按照各自的权重随机选择。'),
-    endpoint: Schema.string().description('safebooru 的 URL。').default('https://safebooru.org/index.php'),
-  })
+  export const Config: Schema<Config> = Schema.intersect([
+    Schema.object({
+      label: Schema.string().default('safebooru').description('图源标签，可用于在指令中手动指定图源。'),
+      weight: Schema.number().min(1).default(1).description('图源权重。在多个符合标签的图源中，将按照各自的权重随机选择。'),
+    }).description('全局设置'),
+    Schema.object({
+      endpoint: Schema.string().description('safebooru 的 URL。').default('https://safebooru.org/index.php'),
```
Comment (MaikoTan) [inline packages/safebooru/src/index.ts:54]:
```suggestion
      endpoint: Schema.string().description('Safebooru 的 URL。').default('https://safebooru.org/index.php'),
```

File: packages/yande/src/index.ts [hunk 5]
```diff
@@ -40,11 +40,15 @@ namespace YandeImageSource {
...
-    weight: Schema.number().min(1).default(1).description('图源权重。在多个符合标签的图源中，将按照各自的权重随机选择。'),
-    endpoint: Schema.string().description('yande 的 URL。').default('https://yande.re'),
-  })
+  export const Config: Schema<Config> = Schema.intersect([
+    Schema.object({
+      label: Schema.string().default('yande').description('图源标签，可用于在指令中手动指定图源。'),
+      weight: Schema.number().min(1).default(1).description('图源权重。在多个符合标签的图源中，将按照各自的权重随机选择。'),
+    }).description('全局设置'),
+    Schema.object({
+      endpoint: Schema.string().description('yande 的 URL。').default('https://yande.re'),
```
Comment (MaikoTan) [inline packages/yande/src/index.ts:49]:
```suggestion
      endpoint: Schema.string().description('Yande.re 的 URL。').default('https://yande.re'),
```

The actual name of this site should be `Yande.re`, as `Yandere` itself is a word but `yande` is not.

**Review by MaikoTan — COMMENTED** [review]
LGTM, just several comments

**Comment by MaikoTan** [comment]
LGTM
@Kabuda-czh you should `approve` this PR then it would get merged automatically.

**Review by Kabuda-czh — APPROVED** [review]

**Pull request merged**